### PR TITLE
Universal Profiling: avoid K/V document delete on duplicates

### PR DESCRIPTION
**Pull request by inge4pres**
## Motivation/summary

The Universal Profiling intake currently generates tombstones when the same stacktrace/stackframe document is ingested from 2 different agents.
We initially understood that an `op_type=index` is a no-op if the document _id is the same and the document has is not changed, but this is not true.

## Checklist

- [X] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [X] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [X] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**? ✅ 
- Is its use being published in **telemetry** to enable product improvement? ⛔ 
- Have system tests been added to avoid regression? ✅ 

## How to test these changes

Enable the profiling intake and send some data through it via an Universal Profiling agent: there should be no deleted docs in the `profiling-stacktraces` and `profiling-stackframes` reading their status via the `_cat` API.

### CODE REVIEW

File: x-pack/apm-server/profiling/collector.go [hunk 1]
```diff
@@ -403,24 +403,24 @@ func (e *ElasticCollector) SetFramesForTraces(ctx context.Context,
...
+				_ error,
 			) {
-				counterStacktracesFailure.Inc()
-				e.logger.With(
-					logp.Error(err),
-					logp.String("error_type", resp.Error.Type),
-				).Errorf("failed to index stacktrace metadata: %s", resp.Error.Reason)
+				// Error is expected here, as we tried to "create" an existing document.
+				// We increment the metric to understand the origin-to-duplicate ratio.
+				counterStacktracesDuplicate.Inc()
```
Comment (axw) [inline x-pack/apm-server/profiling/collector.go:416]: How accurate should this metric be? There might be other errors - should you be parsing the error reason, and only incrementing the counter for specific errors?

Comment (rockdaboot) [reply x-pack/apm-server/profiling/collector.go:416]:
```
const docIDAlreadyExists = "version_conflict_engine_exception"
...
if resp.Error.Type == docIDAlreadyExists {
    counterStacktracesDuplicate.Inc()
    return
}
// log other errors here
```

Comment (inge4pres) [reply x-pack/apm-server/profiling/collector.go:416]: You are right, thanks 🙏🏼 
@rockdaboot shared a suggestion that I'm going to implement here.

**Comment by inge4pres** [comment]
Btw this PR has been labeled with `port-into-apm` because a corresponding change should be applied into apm-server

**Review by rockdaboot — APPROVED** [review]
LGTM

**Review by axw — APPROVED** [review]

**Pull request merged**